metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultDataSource.java

```diff
@@ -67,32 +66,37 @@ public DefaultDataSource(RegistryMonitor registryMonitor, String pollingSettings
         throw new ServiceCombException("bad format servicecomb.metrics.window_time", e);
       }
     }
-    String finalPollingSettings = Strings.join(",", parsePollingSettings.iterator());
-    System.getProperties().setProperty("servo.pollers", finalPollingSettings);
-    List<Long> appliedWindowTimes = getAppliedWindowTime();
-    for (int i = 0; i < appliedWindowTimes.size(); i++) {
-      this.appliedWindowTimes.put(appliedWindowTimes.get(i), i);
+
+    List<Long> sortedPollingSettings = Lists.newArrayList(parsePollingSettings);
+    System.getProperties().setProperty("servo.pollers", Strings.join(",", sortedPollingSettings.iterator()));
```

Comment:
It's not good way to set the system properties here (as it could be overridden anytime), can we setup the servo poller directly?